java/core/src/main/java/com/microsoft/bond/UntaggedProtocolStreamBonded.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import com.microsoft.bond.helpers.ArgumentHelper;
+import com.microsoft.bond.protocol.ProtocolWriter;
+import com.microsoft.bond.protocol.UntaggedProtocolReader;
+
+import java.io.IOException;
+
+/**
+ * An implementation of {@link Bonded} backed by a tagged protocol reader, which effectively
+ * implements delayed deserialization for untagged protocols.
+ *
+ * @param <T> the struct type of this Bonded instance or {@link BondSerializable} if unspecified
+ */
+final class UntaggedProtocolStreamBonded<T extends BondSerializable> extends Bonded<T> {
+
+    private final UntaggedProtocolReader protocolReader;
+    private final StructBondType<T> bondType;
+
+    UntaggedProtocolStreamBonded(UntaggedProtocolReader protocolReader, StructBondType<T> bondType) {
+        this.protocolReader = protocolReader;
+        this.bondType = bondType;
+    }
+
+    @Override
+    public final StructBondType<T> getBondType() {
+        return this.bondType;
+    }
+
+    @Override
+    public void serialize(ProtocolWriter protocolWriter) throws IOException {
+        ArgumentHelper.ensureNotNull(protocolWriter, "protocolWriter");
+        if (this.bondType == null) {
+            throw new InvalidBondDataException("Cannot serialize an unknown struct type within a Bonded instance.");
+        }
+        // deserialize into object using object-backed temporary instance, then serialize it
+        Bonded.fromObject(this.deserialize(), this.bondType).serialize(protocolWriter);
```

Comment:
A temporary transcode. Works for me for now.
What's this code base's style on TODO comments? I expect we'll want to switch this over to transcoding when that is implemented.